packages/lib/src/vis/tiles/TiledHeatmapMesh.tsx

```diff
@@ -1,71 +1,89 @@
 import { useThree } from '@react-three/fiber';
 import { clamp, range } from 'lodash';
+import { useRef } from 'react';
+import type { Group } from 'three';
 
 import { getInterpolator } from '../heatmap/utils';
 import { useCameraState } from '../hooks';
+import type { Size } from '../models';
 import { useAxisSystemContext } from '../shared/AxisSystemProvider';
 import TiledLayer from './TiledLayer';
 import type { TilesApi } from './api';
 import type { ColorMapProps } from './models';
-import { getScaledVisibleDomains } from './utils';
+import { getScaledVisibleBox } from './utils';
 
 interface Props extends ColorMapProps {
   api: TilesApi;
   displayLowerResolutions?: boolean;
   qualityFactor?: number;
+  size?: Size;
 }
 
 function TiledHeatmapMesh(props: Props) {
   const {
     api,
     displayLowerResolutions = true,
     qualityFactor = 1, // 0: Lower quality, less fetch; 1: Best quality
+    size,
     ...colorMapProps
   } = props;
   const { baseLayerIndex, baseLayerSize } = api;
 
   const canvasSize = useThree((state) => state.size);
   const { visSize } = useAxisSystemContext();
+  const meshSize = size ?? visSize;
 
-  const { xVisibleDomain, yVisibleDomain } = useCameraState(
-    (...args) => getScaledVisibleDomains(...args, baseLayerSize),
-    [baseLayerSize]
-  );
+  const groupRef = useRef<Group>(null);
```

Comment:
Just wondering: could we initialize an empty `Object3D` here to somehow end up with no layers without having to deal with an undefined scaled `box`?

Reply:
Not sure what the resulting box would be then.
Returning an empty `Box` makes more sense to me, I'll give it a try and you'll tell what you think of it.

Reply:
Good call, seems the resulting box covers the entire FOV of the camera, because `Object3D` is initialized with a `scale` of (1,1,1).